src/test/java/net/minecraftforge/debug/gameplay/loot/GlobalLootModifiersTest.java

```diff
@@ -0,0 +1,193 @@
+                );
+                event.getRegistry().register(new SmeltingEnchantmentModifier.Serializer().setRegistryName(new ResourceLocation(MODID,"smelting")));
+                event.getRegistry().register(new SilkTouchTestModifier.Serializer().setRegistryName(new ResourceLocation(MODID,"silk_touch_bamboo")));
+            }
+        }
+    }
+
+    private static class SmelterEnchantment extends Enchantment {
+        protected SmelterEnchantment(Rarity rarityIn, EnchantmentType typeIn, EquipmentSlotType[] slots) {
+            super(rarityIn, typeIn, slots);
+        }
+    }
+
+    /**
+     * The smelting enchantment causes this modifier to be invoked, via the smelting loot_modifier json
+     * @author Draco18s
+     *
+     */
+    private static class SmeltingEnchantmentModifier extends LootModifier {
+        public SmeltingEnchantmentModifier(ResourceLocation name, ILootCondition[] conditionsIn) {
+            super(conditionsIn);
+
+        }
+
+        @Override
+        public List<ItemStack> doApply(List<ItemStack> generatedLoot, LootContext context) {
+            ArrayList<ItemStack> ret = new ArrayList<ItemStack>();
+            generatedLoot.forEach((stack) -> ret.add(smelt(stack, context)));
+            return ret;
+        }
+
+        private static ItemStack smelt(ItemStack stack, LootContext context) {
+            return context.getWorld().getRecipeManager().getRecipe(IRecipeType.SMELTING, new Inventory(stack), context.getWorld())
+                    .map(FurnaceRecipe::getRecipeOutput)
+                    .filter(itemStack -> !itemStack.isEmpty())
+                    .map(itemStack -> ItemHandlerHelper.copyStackWithSize(itemStack, stack.getCount() * itemStack.getCount()))
+                    .orElse(stack);
+        }
+
+        private static class Serializer extends LootModifier.Serializer<SmeltingEnchantmentModifier> {
+            @Override
+            public SmeltingEnchantmentModifier read(ResourceLocation name, JsonObject json, ILootCondition[] conditionsIn) {
+                return new SmeltingEnchantmentModifier(name, conditionsIn);
+            }
+        }
+    }
+
+    /**
+     * When harvesting blocks with bamboo, this modifier is invoked, via the silk_touch_bamboo loot_modifier json
+     * @author Draco18s
+     *
+     */
+    private static class SilkTouchTestModifier extends LootModifier {
+        public SilkTouchTestModifier(ResourceLocation name, ILootCondition[] conditionsIn) {
+            super(conditionsIn);
+        }
+
+        @Override
+        public List<ItemStack> doApply(List<ItemStack> generatedLoot, LootContext context) {
+            ItemStack ctxTool = context.get(LootParameters.TOOL);
+            ItemStack fakeTool = ctxTool.copy();
+            fakeTool.addEnchantment(Enchantments.SILK_TOUCH, 1);
+            if(ItemStack.areItemsEqual(ctxTool, fakeTool) && EnchantmentHelper.getEnchantments(ctxTool).containsKey(Enchantments.SILK_TOUCH)) return generatedLoot;
```

Comment:
What is the point of this?

Reply:
This goes back to comments @williewillus made. Boils down to "you can't just get the block's item form."
https://github.com/MinecraftForge/MinecraftForge/pull/5871#issuecomment-542738660

If I've implemented *a test mod* incorrectly, I'm not sure what to say. Its a stand-in for something a modder would do, its not shipped code.

Reply:
Test mods also are used as examples, and unfortunately copy/pasted by idiot modders.
Perhaps this particular modifier would be better served as a pre-processor, something that can always add the enchantment for silk touch to the context? Instead of trying to figure out how to reproduce something after the fact.

Reply:
LootContext objects are `private final`, so there's no way to "add" to it. Yes, it could be modified with reflection, but I don't think that's the appropriate action to take.